src/ACEtk/block/SecondaryParticleProductionBlock/src/generateXSS.hpp

```diff
@@ -0,0 +1,24 @@
+static std::vector< double > generateXSS(
+    std::size_t index,
+    std::vector< double >&& production,
+    std::vector< double >&& heating ) {
+
+  std::size_t size = production.size();
+  if ( heating.size() != size ) {
```

Comment:
Interesting that there is a specific check for size here in this block. This does not occur elsewhere. Is there a reason? If so, is it better to check it in the `verifySize.hpp` check once?

Reply:
verifySize is only used for checking the iterator based constructor - which we use to construct from an existing XSS array. generateXSS is only called when constructing from scratch.